salt/cloud/clouds/gce.py

```diff
@@ -2616,6 +2623,9 @@ def update_pricing(kwargs=None, call=None):
     url = 'https://cloudpricingcalculator.appspot.com/static/data/pricelist.json'
     price_json = http.query(url, decode=True, decode_type='json')
 
+    if not os.path.exists(__opts__['cachedir']):
```

Comment:
I don't think that a driver is the right place for this. If we need to validate an environment, IMHO it should be doing higher up the stack instead of inside the driver.